packages/grid_client/scripts/mass_deployments.ts

```diff
@@ -1,21 +1,183 @@
+    );
+  try {
+    events.emit("logs", "Validating workloads");
+    await grid3.machines.twinDeploymentHandler.validate(twinDeployments);
+    await grid3.machines.twinDeploymentHandler.checkNodesCapacity(twinDeployments);
+    await grid3.machines.twinDeploymentHandler.checkFarmIps(twinDeployments);
+  } catch (error) {
+    events.emit("logs", "Error Validating workloads:" + error);
+  }
+
+  const contracts: DeploymentResultContracts = {
+    created: [],
+    updated: [],
+    deleted: [],
+  };
+  const resultContracts: DeploymentResultContracts = {
+    created: [],
+    updated: [],
+    deleted: [],
+  };
+  let nodeExtrinsics: ExtrinsicResult<Contract>[] = [];
+  let nameExtrinsics: ExtrinsicResult<Contract>[] = [];
+  let deletedExtrinsics: ExtrinsicResult<number>[] = [];
+
+  for (const twinDeployment of twinDeployments) {
+    for (const workload of twinDeployment.deployment.workloads) {
+      if (!twinDeployment.network) break;
+      if (workload.type === WorkloadTypes.network || workload.type === WorkloadTypes.networklight) {
+        events.emit("logs", `Updating network workload with name: ${workload.name}`);
+        twinDeployment.network.updateWorkload(twinDeployment.nodeId, workload);
+      }
+    }
+
+    const extrinsics = await grid3.machines.twinDeploymentHandler.PrepareExtrinsic(twinDeployment, contracts);
+    nodeExtrinsics = nodeExtrinsics.concat(extrinsics.nodeExtrinsics);
+    nameExtrinsics = nameExtrinsics.concat(extrinsics.nameExtrinsics);
+    deletedExtrinsics = deletedExtrinsics.concat(extrinsics.deletedExtrinsics);
+  }
+
+  const extrinsicResults: Contract[] = await grid3.machines.twinDeploymentHandler.tfclient.applyAllExtrinsics<Contract>(
+    [...nodeExtrinsics, ...nameExtrinsics],
+  );
+
+  for (const contract of extrinsicResults) {
+    const updatedContract = contracts.updated.find(c => c["contractId"] === contract.contractId);
+    if (!updatedContract) contracts.created.push(contract);
+  }
+
+  const successfulNodes = new Set<number>();
+  const failedNodes = new Set<number>();
+  for (const twinDeployment of twinDeployments) {
+    try {
+      if (twinDeployment.operation === Operations.deploy) {
+        events.emit("logs", `Sending deployment to node_id: ${twinDeployment.nodeId}`);
+        for (const contract of extrinsicResults) {
+          if (twinDeployment.deployment.challenge_hash() === contract.contractType.nodeContract.deploymentHash) {
+            twinDeployment.deployment.contract_id = contract.contractId;
+            if (
+              twinDeployment.returnNetworkContracts ||
+              !(
+                twinDeployment.deployment.workloads.length === 1 &&
+                twinDeployment.deployment.workloads[0].type === WorkloadTypes.network
+              )
+            )
+              resultContracts.created.push(contract);
+            break;
+          }
+        }
```

Comment:
can you explain what are we doing here, i can see you are updating the resultContracts but its not clear the meaning of inner conditionse ?